engine/development/deploymentManager/src/main/scala/pl/touk/nussknacker/development/manager/DevelopmentDeploymentManagerProvider.scala

```diff
@@ -56,22 +56,35 @@ class DevelopmentDeploymentManager(actorSystem: ActorSystem) extends DeploymentM
   }
 
   override def validate(processVersion: ProcessVersion, deploymentData: DeploymentData, canonicalProcess: CanonicalProcess): Future[Unit] = {
-    if (canonicalProcess.metaData.additionalFields.flatMap(_.description).contains("fail")) {
+    if (description(canonicalProcess).contains("fail")) {
       Future.failed(new IllegalArgumentException("Scenario validation failed as description contains 'fail'"))
     } else {
       Future.successful(())
     }
   }
 
+  private def description(canonicalProcess: CanonicalProcess) = {
+    canonicalProcess.metaData.additionalFields.flatMap(_.description)
+  }
+
   override def deploy(processVersion: ProcessVersion, deploymentData: DeploymentData, canonicalProcess: CanonicalProcess, savepointPath: Option[String]): Future[Option[ExternalDeploymentId]] = {
     logger.debug(s"Starting deploying scenario: ${processVersion.processName}..")
+    val previous = memory.get(processVersion.processName)
     val duringDeployStateStatus = createAndSaveProcessState(DuringDeploy, processVersion)
     val result = Promise[Option[ExternalDeploymentId]]()
     actorSystem.scheduler.scheduleOnce(sleepingTimeSeconds, new Runnable {
       override def run(): Unit = {
         logger.debug(s"Finished deploying scenario: ${processVersion.processName}.")
-        result.complete(Success(duringDeployStateStatus.deploymentId))
-        asyncChangeState(processVersion.processName, Running)
+        if (description(canonicalProcess).contains("deployFail")) {
```

Comment:
from where comes this "deployFail" string ?

Reply:
from here - I can extract it to constant

Reply:
Ok now I get how it works